chore: clone initial project from existing repository

fredysiswanto opened the pull request:
## Description
This PR clones the initial codebase from [source repository](https://github.com/fredysiswanto/template-api-testing-use-jest) into this new repository.  
It includes all relevant code, folder structures, and configurations needed to continue development here.


##Key Features:

Helper function generator to simplify API calls
Test case generator for API endpoints
Modular and scalable project structure
Built-in CI using GitHub Actions
how to use https://www.youtube.com/watch?v=bLxLptxJ0ew

fredysiswanto committed:
clone repo from https://github.com/fredysiswanto/template-api-testing-use-jest

fredysiswanto committed:
Refactor API structure and update test paths

- Updated Jest and JS config paths for test and API helper files.
- Changed file generation paths in CLI script to reflect new structure.
- Removed deprecated auth, product, and user API files along with their corresponding tests and templates.
- Updated README to reflect changes in file generation locations.

fredysiswanto committed:
Merge pull request #1 from fredysiswanto/dev

Refactor API structure and update test paths

fredysiswanto committed:
Update GitHub Actions workflow to trigger on 'main' branch instead of 'master'

lamhotsiagian commented on config/helper.js:
```diff
@@ -0,0 +1,19 @@
+const { HEADERS } = require('./config');
+// Fungsi untuk menggabungkan header dan menangani debugging
```
@fredysiswanto could you please update the comment using en language

lamhotsiagian commented on config/config.js:
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  BASE_URL: 'https://fakestoreapi.com', // Ganti dengan URL API asli
```
please use en language

lamhotsiagian commented on db.json:
```diff
@@ -0,0 +1,63 @@
+{
```
please use en language

lamhotsiagian reviewed (changes requested):
Hi mas @fredysiswanto thank you for your PR, left some comments. 

- please change all comment to english in all files
- remove this file [config/logger.js](https://github.com/Test-Architect/jest-js-sample/pull/1/files#diff-386c505d3fbf272fd824783e8824029b1012d0c24b81b04ead6e795197412e71)
- remove this file [sample_report.png](https://github.com/Test-Architect/jest-js-sample/pull/1/files#diff-e741e642849ec771c9d4195add4810be5dc39b8ca9457ba2b70f6d5967da4ed6)
- remove this file [.github/workflows/node.js.yml](https://github.com/Test-Architect/jest-js-sample/pull/1/files#diff-014228303dff9a1af15f4bbd18401f906380129b10ae2a2c62f8b8be592ff88e)

fredysiswanto committed:
Remove unused files and update comments for clarity

- Deleted sample_report.png and logger.js as they are no longer needed.
- Removed the Node.js CI workflow file.
- Updated comments in various files for better understanding and consistency.
- Added sample generation comments in product.js and user.js files.

fredysiswanto commented:
> Hi mas @fredysiswanto thank you for your PR, left some comments.
> 
> * please change all comment to english in all files
> * remove this file [config/logger.js](https://github.com/Test-Architect/jest-js-sample/pull/1/files#diff-386c505d3fbf272fd824783e8824029b1012d0c24b81b04ead6e795197412e71)
> * remove this file [sample_report.png](https://github.com/Test-Architect/jest-js-sample/pull/1/files#diff-e741e642849ec771c9d4195add4810be5dc39b8ca9457ba2b70f6d5967da4ed6)
> * remove this file [.github/workflows/node.js.yml](https://github.com/Test-Architect/jest-js-sample/pull/1/files#diff-014228303dff9a1af15f4bbd18401f906380129b10ae2a2c62f8b8be592ff88e)

Hi mas bro @lamhotsiagian.
Done. Fixed last commit

lamhotsiagian reviewed (approved):
thank you
